pkg/registry/apis/dashboard/register.go

```diff
@@ -144,42 +183,365 @@ func (b *DashboardsAPIBuilder) InstallSchema(scheme *runtime.Scheme) error {
+			return nil
 		}
 
-		if deleteOptions.GracePeriodSeconds == nil || *deleteOptions.GracePeriodSeconds != 0 {
-			nsInfo, err := claims.ParseNamespace(a.GetNamespace())
-			if err != nil {
-				return fmt.Errorf("%v: %w", "failed to parse namespace", err)
-			}
+		return fmt.Errorf("%v: %w", "delete hook failed to check if dashboard is provisioned", err)
+	}
 
-			provisioningData, err := b.dashboardProvisioningService.GetProvisionedDashboardDataByDashboardUID(ctx, nsInfo.OrgID, a.GetName())
-			if err != nil {
-				if errors.Is(err, dashboards.ErrProvisionedDashboardNotFound) ||
-					errors.Is(err, dashboards.ErrDashboardNotFound) ||
-					apierrors.IsNotFound(err) {
-					return nil
-				}
+	if provisioningData != nil {
+		return apierrors.NewBadRequest(dashboards.ErrDashboardCannotDeleteProvisionedDashboard.Reason)
+	}
 
-				return fmt.Errorf("%v: %w", "delete hook failed to check if dashboard is provisioned", err)
-			}
+	return nil
+}
 
-			if provisioningData != nil {
-				return apierrors.NewBadRequest(dashboards.ErrDashboardCannotDeleteProvisionedDashboard.Reason)
-			}
+// validateCreate validates dashboard creation
+func (b *DashboardsAPIBuilder) validateCreate(ctx context.Context, a admission.Attributes, o admission.ObjectInterfaces) error {
+	// Get the dashboard object
+	dashObj := a.GetObject()
+
+	title, refresh, err := getDashboardProperties(dashObj)
+	if err != nil {
+		return fmt.Errorf("error extracting dashboard properties: %w", err)
+	}
+
+	accessor, err := utils.MetaAccessor(dashObj)
+	if err != nil {
+		return fmt.Errorf("error getting meta accessor: %w", err)
+	}
+
+	// Parse namespace for orgID
+	nsInfo, err := types.ParseNamespace(a.GetNamespace())
+	if err != nil {
+		return fmt.Errorf("failed to parse namespace: %w", err)
+	}
+
+	// Basic validations
+	if err := b.validateBasicProperties(title, accessor); err != nil {
+		return err
+	}
+
+	// Check for UID uniqueness
+	// Let the storage handle this.
+	// if accessor.GetName() != "" {
+	// 	existing, err := b.dashStore.GetDashboard(ctx, &dashboards.GetDashboardQuery{
+	// 		UID:   accessor.GetName(),
+	// 		OrgID: nsInfo.OrgID,
+	// 	})
+	// 	if err == nil && existing != nil {
+	// 		return dashboards.ErrDashboardWithSameUIDExists
+	// 	} else if err != nil && !errors.Is(err, dashboards.ErrDashboardNotFound) {
+	// 		return fmt.Errorf("error checking dashboard UID uniqueness: %w", err)
+	// 	}
+	// }
+
+	// Validate folder existence if specified
+	if accessor.GetFolder() != "" {
+		if err := b.validateFolderExists(ctx, accessor.GetFolder(), nsInfo.OrgID); err != nil {
+			return err
+		}
+	}
+
+	// Validate refresh interval
+	if err := b.validateRefreshInterval(refresh); err != nil {
+		return err
+	}
+
+	// Check if dashboard is provisioned and if it allows updates
+	//mgr, hasMgr := accessor.GetManagerProperties()
+	//if hasMgr && mgr.Identity != "" && !mgr.AllowsEdits {
+	//	return dashboards.ErrDashboardCannotSaveProvisionedDashboard
+	//}
```

Comment:
This will be handled in the storage strategy CC @ryantxu

Reply:
Yeah provisioned storage is something I excluded in this PR. But will the storage strategy also handle legacy?

Reply:
> Yeah provisioned storage is something I excluded in this PR. But will the storage strategy also handle legacy?

For legacy this is IIRC already handled in the legacy storage, but I might be wrong – @stephaniehingtgen / @ryantxu should know better.